pkg/pidfile/pidfile.go

```diff
@@ -0,0 +1,51 @@
+package pidfile
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+)
+
+type PidFile struct {
+	path string
+}
+
+func checkPidFileAlreadyExists(path string) error {
+	if pidString, err := ioutil.ReadFile(path); err == nil {
+		pid, err := strconv.Atoi(string(pidString))
+		if err == nil {
+			if _, err := os.Stat(fmt.Sprintf("/proc/%d/", pid)); err == nil {
```

Comment:
That sounds cleaner, indeed. However, I wonder to what extent this actually is platform independent...